samtranslator/model/sam_resources.py

```diff
@@ -802,6 +803,114 @@ def _validate_deployment_preference_and_add_update_policy(
             )
 
 
+class SamCanary(SamResourceMacro):
+    """SAM canary macro."""
+
+    resource_type = "AWS::Serverless::Canary"
+    property_types = {
+        "FunctionName": PropertyType(True, is_str()),
+        "Handler": PropertyType(True, is_str()),
+        "Runtime": PropertyType(True, is_str()),
+        "CodeUri": PropertyType(False, one_of(is_str(), is_type(dict))),
+        "InlineCode": PropertyType(False, one_of(is_str(), is_type(dict))),
+        "MemorySize": PropertyType(False, is_type(int)),
+        "Tags": PropertyType(False, is_type(dict)),
+        "Tracing": PropertyType(False, is_type(bool)),
+        "Timeout": PropertyType(False, is_type(int)),
+        "Role": PropertyType(True, is_str()),
+        "Schedule": PropertyType(True, one_of(is_type(dict), is_str())),
+        "StartCanaryAfterCreation": PropertyType(True, is_type(bool)),
+        "ArtifactS3Location": PropertyType(True, one_of(is_type(dict), is_str())),
+        "FailureRetentionPeriod": PropertyType(False, one_of(is_type(dict), is_str())),
+        "SuccessRetentionPeriod": PropertyType(False, one_of(is_type(dict), is_str())),
+        "VpcConfig": PropertyType(False, is_type(dict)),
+        "Environment": PropertyType(False, dict_of(is_str(), is_type(dict))),
+    }
+
+    def to_cloudformation(self, **kwargs):
+        """Returns the Synthetics Canary to which this SAM Canary corresponds.
+
+        :param dict kwargs: already-converted resources that may need to be modified when converting this \
+        macro to pure CloudFormation
+        :returns: a list of vanilla CloudFormation Resources, to which this Function expands
+        :rtype: list
+        """
+        resources = []
+        syntheticsCanary = self._construct_synthetics_canary()
+        resources.append(syntheticsCanary)
+
+        return resources
+
+    def _construct_synthetics_canary(self):
+        """Constructs a AWS::Synthetics::Canary resource."""
+        canary = SyntheticsCanary(
+            self.logical_id, depends_on=self.depends_on, attributes=self.get_passthrough_resource_attributes()
+        )
+        canary.ArtifactS3Location = self.ArtifactS3Location
+        canary.Code = self._construct_code_dict()
+        canary.ExecutionRoleArn = self.Role
```

Comment:
Just adding a comment in case there are other eyes on PR. We'll add support for auto creation of Role in next PRs.